tests/utils.go

```diff
@@ -268,6 +266,12 @@ type WarningsPolicy struct {
 	WarningsIgnoreList []string
 }
 
+func TempDir() string {
```

Comment:
I am not convinced that adding a function to utils is any better than keeping a const in it.
We should not add anything to this file IMO.

In addition, hiding the error using an assertion in helpers is not recommended, it messes up the assertion log. Using an offset assertion is possible, but that may not help because we do not control the call hops from the caller.

BTW, the recommended function to use now is `os.MkdirTemp`.

Reply:
> I am not convinced that adding a function to utils is any better than keeping a const in it.

I tried to explain this in the PR text. When you expose a const, you implicitly state that the content of the const is important, that it may be concatenated, that it is ok to use it for something other than creating a temp dir. These are all false, hence the const should be hidden.

> We should not add anything to this file IMO.

I dig you. `utils` files are evil junkyards. However, if anything should be placed in a utils file, is something like this generic dir-creation function. With this change I'm not making the file very much longer, but I improve its API. I hope we can think about another location for this function in another time.

> 
> In addition, hiding the error using an assertion in helpers is not recommended, it messes up the assertion log. Using an offset assertion is possible, but that may not help because we do not control the call hops from the caller.

Not sure I understand your point about adding `Expect` to a util function. Maybe its my Python state-of-mind, craving for exceptions here in GoLang. Moving the expectation into the function made all the users shorter and simpler. The down side is that `TempDir` is not usable outside this test framework, but it is quite specific anyway.

> 
> BTW, the recommended function to use now is `os.MkdirTemp`.

Right. GoLang people agree that `ioutil` is an evil junkyard.

Reply:
> Not sure I understand your point about adding Expect to a util function. Maybe its my Python state-of-mind, craving for exceptions here in GoLang. Moving the expectation into the function made all the users shorter and simpler. The down side is that TempDir is not usable outside this test framework, but it is quite specific anyway.

`Expect` is supposed to be called in the tests themselves, so their error logging will include the location where it happened (line number, filename etc..).
If it asserts in a different place in the stack, it will give a wrong number, which has a potential to make it hard for one to find the actual caller. This is fixable only by using `ExpectWithOffset` which is problematic (you do not really know the offset if this is placed in a generic util).

> I hope we can think about another location for this function in another time.

I do not like it, but it is not a blocker.

> I tried to explain this in the PR text. When you expose a const, you implicitly state that the content of the const is important, that it may be concatenated, that it is ok to use it for something other than creating a temp dir. These are all false, hence the const should be hidden.

I am good with this reasoning, but in the balance of it all, I find it no better (and no worse).
In my view, we replace one problematic thing with another.
I do not like it as well, but this one is also not a blocker.